### ISSUE TITLE
Safety Certified FreeRTOS

### ISSUE THREAD

**Issue by amazonKamath** [FreeRTOS/FreeRTOS-Kernel#906]
The FreeRTOS kernel will be assessed, and certified for functional safety compliance under IEC 61508. With the safety-compliant kernel, FreeRTOS developers will be able to more easily build and test systems that require safety certification. The kernel, matched with an MCU, provides a foundation upon which applications, devices, and systems can be built following functional safety standards and certified by authorized vendors. IEC 61508 is a base functional safety standard used across industries, and upon which other standards are derived to meet the safety needs of industrial, automotive, robotics, medical, and other applications.

**Comment by BerndThie** [comment]
This is a very interesting topic! 
When can we expect more information?

**Comment by archigup** [comment]
Hi, this is actively being worked on. Is there particular information you are interested in?

**Comment by BerndThie** [comment]
Thanks for the fast reply! I'm firstly interested in the SIL you are targeting and the timeline. When do you expect to go public with the safety certified FreeRTOS?

**Comment by joshzarr** [comment]
Thank you for your interest in FreeRTOS!

We are currently working to certify FreeRTOS to IEC 61508 Systematic Capability 3 for use in systems up to SIL 3. Our plan is to have separate certificates for the FreeRTOS Kernel and individual FreeRTOS Ports. We are targeting submission of all documentation and test artifacts to our assessor for Q1 2025 and anticipate the certificate being granted in Q2 2025. Our first safety certified port will target the TI Hercules series of microcontrollers.

Once we have completed the certification process we will follow on with further announcements regarding the general availability of Safety Certified FreeRTOS.

**Comment by OndrejMasopust** [comment]
Hello,
is this somehow different from the already available SAFERTOS? Will the certificates with the source code be available for free or paid?

**Comment by amazonKamath** [comment]
> Hello, is this somehow different from the already available SAFERTOS? Will the certificates with the source code be available for free or paid?

Yes, this is a separate certification effort from SAFERTOS, based on FreeRTOS kernel v11. We intend to keep the safety certified code free and open source under the MIT license. A Safety Certification Documentation Bundle to help developers complete certification of their own products is being considered. We do not have pricing guidance for the Certification Bundle at this time.

**Comment by FlavioBGefran** [comment]
Indeed, It's a really interesting topic !

Have you planned to certify the freertos port to support a generic Cortex-M (armv7m and armv8m) mcu, based on the MPU feature?
 [https://www.freertos.org/Security/04-FreeRTOS-MPU-memory-protection-unit](url)

Thank you very much for your work.

Best regards,

Flavio

**Comment by cvh2bp** [comment]
> Thank you for your interest in FreeRTOS!
> 
> We are currently working to certify FreeRTOS to IEC 61508 Systematic Capability 3 for use in systems up to SIL 3. Our plan is to have separate certificates for the FreeRTOS Kernel and individual FreeRTOS Ports. We are targeting submission of all documentation and test artifacts to our assessor for Q1 2025 and anticipate the certificate being granted in Q2 2025. Our first safety certified port will target the TI Hercules series of microcontrollers.
> 
> Once we have completed the certification process we will follow on with further announcements regarding the general availability of Safety Certified FreeRTOS.

Hello,

This is a really interesting topic for us as well. Do you have any updates about the mentioned certificate process for the Kernel? Do you have any information whether a certified port for TI Sitara series (Cortex R5F and C7x DSP cores) is also planned? If yes is it possible to request a timeline for that?

Thanks and Best Regards,
Viktor

**Comment by AIAsensus** [comment]
Hello,
Thank you for your work on FreeRTOS. This is a very interesting topic indeed. Is there any update on the timeline for the release of the safety certified FreeRTOS Kernel and potentially the Safety Certification Documentation Bundle? 
How about the individual ports? Is there a plan/timeline for a certified port for an ARM Cortex-M4?
Thank you very much for your time!
Alice

**Comment by kstribrnAmzn** [comment]
The Kernel certification effort is currently ongoing. All of the kernel documents and artifacts have been generated and are going through final review. This review is being done by @joshzarr and myself with the final approval being given by our assessor. I cannot give exact dates for when these documents will be signed off on given we do not perform the assessment ourselves but our team has been striving towards a late Q4 2025 certification. This certification will include the Kernel as well as the TI Hercules port as mentioned previously.

We do plan on growing our certified feature set as time goes on (MPU, additional ports, etc). We will base the order of features added off of end user needs. The split certification strategy will allow us to certify additional ports significantly faster than this initial certification (I'd expect in the range of a few months).

We will post more information on how to obtain the certification bundle at a later date. We are currently discussing how we would like to distribute the materials.

**Comment by kstribrnAmzn** [comment]
We have received safety certification ([kernel](https://fs-products.tuvasi.com/certificates?cert_id=12090), [port](https://fs-products.tuvasi.com/certificates?cert_id=13542)). Discussions regarding the certification bundle have resumed again. I cannot provide a date yet but felt the community deserves an update.

**Comment by FlavioBGefran** [comment]
Thanks for the update, KstribrnAmzn. It's great news!

We look forward to the certified port for ARMv8 and the ARMv7 port.

Thanks again for your work.

Flavio
